README.md

```diff
@@ -1,2 +1,111 @@
 # pshenmic-dev-dfo
-The homepage of the pshenmic.dev DFO source files
+
+The homepage of the pshenmic.dev DFO source files.
+
+## Data Contract Structure
+
+This project includes three key documents in the data contract schema: **Project**, **Tasks**, and **Claim**. Below is a detailed description of each.
+
+### Project Document
+
+The `Project` document defines the structure for storing project information.
+
+- **name** (string, required):  
+  - Position: 0  
+  - Description: The name of the project.  
+  - Maximum Length: 63 characters.
+  
+- **description** (string):  
+  - Position: 1  
+  - Description: A detailed description of the project.  
+  - Maximum Length: 1000 characters.
+  
+- **url** (string):  
+  - Position: 2  
+  - Description: The URL associated with the project.  
+  - Maximum Length: 255 characters.
+
+#### Required Fields
+- `name`
+- `$createdAt`
+- `$updatedAt`
+
+Additional properties are not allowed.
+
+---
+
+### Tasks Document
+
+The `Tasks` document is used to define individual tasks within a project.
+
+- **title** (string, required):  
+  - Position: 0  
+  - Description: The title of the task.  
+  - Maximum Length: 63 characters.
+  
+- **description** (string):  
+  - Position: 1  
+  - Description: A detailed description of the task.  
+  - Maximum Length: 1000 characters.
+  
+- **url** (string):  
+  - Position: 2  
+  - Description: A URL with further details about the task.  
+  - Maximum Length: 255 characters.
+  
+- **assignee** (array):  
+  - Position: 3  
+  - Description: The identity of the assignee who will execute the task.  
+  - Byte Array: true  
+  - Length: 32 bytes.
+  
+- **projectId** (string, required):  
```

Comment:
should be byte array, since we will be linking to the project document identifier